modules/examples/sources/msg-generator/msg-generator.c

```diff
@@ -92,9 +93,8 @@ msg_generator_sd_new(GlobalConfig *cfg)
 {
   MsgGeneratorSourceDriver *self = g_new0(MsgGeneratorSourceDriver, 1);
   log_src_driver_init_instance(&self->super, cfg);
-
   msg_generator_source_options_defaults(&self->source_options);
-
+  self->source_options.name_value = g_hash_table_new_full(g_str_hash, g_str_equal, NULL, NULL);
```

Comment:
You need to specify the destroy functions here, specifyng the destructor function for the keys and values: `g_free` and `log_template_unref`